src/procedures/DisableController.ts

```diff
@@ -0,0 +1,56 @@
+import { Procedure } from './Procedure';
+import {
+  ProcedureType,
+  PolyTransactionTag,
+  DisableControllerProcedureArgs,
+  ErrorCode,
+} from '../types';
+import { PolymathError } from '../PolymathError';
+
+export class DisableController extends Procedure<DisableControllerProcedureArgs> {
+  public type = ProcedureType.DisableController;
+
+  public async prepareTransactions() {
+    const { signature, symbol } = this.args;
+    const { contractWrappers, currentWallet } = this.context;
+
+    let securityToken;
+
+    try {
+      securityToken = await contractWrappers.tokenFactory.getSecurityTokenInstanceFromTicker(
+        symbol
+      );
+    } catch (err) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `There is no Security Token with symbol ${symbol}`,
+      });
+    }
+
+    const [owner, account] = await Promise.all([securityToken.owner(), currentWallet.address()]);
+
+    if (account !== owner) {
+      throw new PolymathError({
+        code: ErrorCode.ProcedureValidationError,
+        message: `You must be the owner of this Security Token to disable the controller`,
+      });
+    }
+
+    if (!(await securityToken.isControllable())) {
```

Comment:
Ideally we don't use await inside conditionals or other expressions (it's just a readability thing).

Store the result of this call in a variable and then use it inside the if condition